lib/ssl/ssl3con.c

```diff
@@ -10371,15 +9664,14 @@ ssl3_HandleCertificateVerify(sslSocket *ss, SSL3Opaque *b, PRUint32 length,
         goto alert_loser;
     }
 
-    if (ss->ssl3.hs.hashType != handshake_hash_record) {
-        if (!hashes) {
-            PORT_Assert(0);
-            desc = internal_error;
-            errCode = SEC_ERROR_LIBRARY_FAILURE;
-            goto alert_loser;
-        }
-        hashesForVerify = hashes;
-    } else {
+    if (!hashes) {
```

Comment:
If this is the right change (and I think that it is), then I think that we can simply add a PORT_Assert on line 9657.